app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt

```diff
@@ -99,31 +99,73 @@ class TheActivityControllerTest {
   }
 
   @Test
-  @Parameters(value = [
-    "NOT_LOGGED_IN|false",
-    "OTP_REQUESTED|true",
-    "LOGGED_IN|true",
-    "RESETTING_PIN|false",
-    "RESET_PIN_REQUESTED|true"
-  ])
-  fun `when activity is started, user is logged in and user was inactive then app lock should be shown`(
-      loggedInStatus: User.LoggedInStatus,
-      shouldShowAppLock: Boolean
-  ) {
+  fun `when activity is started, user has requested an OTP, and user was inactive then app lock should be shown`() {
     whenever(userSession.isUserLoggedIn()).thenReturn(true)
     whenever(userSession.loggedInUser())
-        .thenReturn(Observable.just(Just(TestData.loggedInUser(loggedInStatus = loggedInStatus, status = UserStatus.ApprovedForSyncing))))
+        .thenReturn(Observable.just(Just(TestData.loggedInUser(loggedInStatus = OTP_REQUESTED, status = UserStatus.ApprovedForSyncing))))
 
     val lockAfterTime = currentTimestamp.minusSeconds(TimeUnit.MINUTES.toSeconds(1))
     whenever(lockAfterTimestamp.get()).thenReturn(lockAfterTime)
 
     uiEvents.onNext(Started(null))
 
-    if (shouldShowAppLock) {
-      verify(activity).showAppLockScreen()
-    } else {
-      verify(activity, never()).showAppLockScreen()
-    }
+    verify(activity).showAppLockScreen()
+  }
+
+  @Test
+  fun `when activity is started, user is logged in, and user was inactive then app lock should be shown`() {
+    whenever(userSession.isUserLoggedIn()).thenReturn(true)
+    whenever(userSession.loggedInUser())
+        .thenReturn(Observable.just(Just(TestData.loggedInUser(loggedInStatus = LOGGED_IN, status = UserStatus.ApprovedForSyncing))))
+
+    val lockAfterTime = currentTimestamp.minusSeconds(TimeUnit.MINUTES.toSeconds(1))
+    whenever(lockAfterTimestamp.get()).thenReturn(lockAfterTime)
+
+    uiEvents.onNext(Started(null))
+
+    verify(activity).showAppLockScreen()
+  }
+
+  @Test
+  fun `when activity is started, user has requested a PIN reset, and user was inactive then app lock should be shown`() {
+    whenever(userSession.isUserLoggedIn()).thenReturn(true)
+    whenever(userSession.loggedInUser())
+        .thenReturn(Observable.just(Just(TestData.loggedInUser(loggedInStatus = RESET_PIN_REQUESTED, status = UserStatus.ApprovedForSyncing))))
+
+    val lockAfterTime = currentTimestamp.minusSeconds(TimeUnit.MINUTES.toSeconds(1))
+    whenever(lockAfterTimestamp.get()).thenReturn(lockAfterTime)
+
+    uiEvents.onNext(Started(null))
+
+    verify(activity).showAppLockScreen()
+  }
+
+  @Test
+  fun `when activity is started, user is not logged in and user was inactive then app lock should not be shown`() {
+    whenever(userSession.isUserLoggedIn()).thenReturn(true)
```

Comment:
Shouldn't this return `false`?

Reply:
Good catch. Also will need to check why the test did not break.

Reply:
Ok, I just checked and the test is correct. The problem is that `UserSession#isUserLoggedIn` method is badly named.

What it actually does is check if there is a user present locally or not, but not if they are _actually_ logged in. This is handled when creating the user for the test with a `loggedInStatus` of `NOT_LOGGED_IN`.

Created a [ticket](https://app.clubhouse.io/simpledotorg/story/835/rename-badly-named-method-in-usersession) to track renaming this.